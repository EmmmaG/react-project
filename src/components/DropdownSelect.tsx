import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import OutlinedInput from '@material-ui/core/OutlinedInput'
import Typography from '@material-ui/core/Typography'
import KeyboardArrowDown from '@material-ui/icons/KeyboardArrowDown'

const styles = () => ({
	selectContainer: {
		textAlign: 'left' as 'left',
		marginTop: '10px',
		marginBottom: '20px',
		marginRight: '20px',
		display: 'inline-block',
	},
	select: {
		height: '35px',
		backgroundColor: 'white',
	},
	selectRoot: {
		display: 'flex',
		alignItems: 'center',
	},
	title: {
		marginBottom: '2px',
	},
})

export interface Select {
	value: string
	text: string
}

export interface Props extends WithStyles<typeof styles> {
	title: string
	onChange: (selectedValue: string) => void
	value: string
	options: Select[]
	width: string
}

export class DropdownSelect extends React.Component<Props> {
	private handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
		this.props.onChange(event.target.value)
	}

	private renderOptions = () => this.props.options.map(o => (
		<option value={o.value} key={o.value}>
			{o.text}
		</option>
	))

	public render() {
		const { classes, value, title, width } = this.props

		return (
			<div className={classes.selectContainer}>
				<Typography className={classes.title}>{title}</Typography>
				<FormControl variant='outlined'>
					<Select
						style={{ width }}
						className={classes.select}
						classes={{ root: classes.selectRoot }}
						native
						value={value}
						onChange={this.handleChange}
						input={
							<OutlinedInput
								name={title}
								labelWidth={0}
								id={title}
							/>
						}
						IconComponent={() => (
							<KeyboardArrowDown />
						)}>
					>
						{this.renderOptions()}
					</Select>
				</FormControl>
			</div>
		)
	}
}

export default withStyles(styles)(DropdownSelect)
