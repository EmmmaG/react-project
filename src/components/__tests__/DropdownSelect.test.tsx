import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import DropdownSelect, { Props } from '../DropdownSelect'

const props: Props = {
	title: 'select',
	onChange: jest.fn(),
	value: '1',
	options: [{
		value: '1',
		text: 'first',
	}, {
		value: '2',
		text: 'second',
	}],
	width: '200px',
	classes: idObj,
}

describe('DropdownSelect', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<DropdownSelect {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
