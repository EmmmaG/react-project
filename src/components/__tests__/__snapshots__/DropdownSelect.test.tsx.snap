// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownSelect renders correctly 1`] = `
<DropdownSelect
  classes={
    Object {
      "select": "DropdownSelect-select-2",
      "selectContainer": "DropdownSelect-selectContainer-1",
      "selectRoot": "DropdownSelect-selectRoot-3",
      "title": "DropdownSelect-title-4",
    }
  }
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "text": "first",
        "value": "1",
      },
      Object {
        "text": "second",
        "value": "2",
      },
    ]
  }
  title="select"
  value="1"
  width="200px"
/>
`;
